closed/src/java.base/share/classes/com/sun/crypto/provider/NativeGaloisCounterMode.java

```diff
@@ -343,40 +343,47 @@ int encrypt(byte[] in, int inOfs, int len) {
      */
     int encryptFinal(byte[] in, int inOfs, int len, byte[] out, int outOfs)
         throws IllegalBlockSizeException, ShortBufferException {
+        byte[] keyCopy;
+        byte[] ivCopy;
+        byte[] aad;
+        synchronized (this) {
+            if (len > MAX_BUF_SIZE - tagLenBytes) {
```

Comment:
A local copy of `tagLenBytes` should be created so that value can be used after the `synchronized` block.

Reply:
I added a local copy of `tagLenBytes`